graphql_compiler/compiler/compiler_frontend.py

```diff
@@ -393,15 +402,20 @@ def _compile_vertex_ast(schema, current_schema_type, ast,
         # This will need to change if nested @optionals have to be supported.
         within_optional_scope = 'optional' in context and not edge_traversal_is_optional
 
+        # Extracting the last inserted MarkLocation
+        current_marked_location = context['marked_location_stack'][-1]
```

Comment:
Let's inline this so it's obvious that the check below uses the context, and that later on, the mutation affects an element on the stack.